service/src/main/java/bio/terra/cbas/runsets/types/CbasArray.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.cbas.runsets.types;
+
+import bio.terra.cbas.model.ParameterTypeDefinition;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CbasArray implements CbasOptional {
+
+  private final List<CbasValue> values;
+
+  public CbasArray(List<CbasValue> values) {
+    this.values = values;
+  }
+
+  @Override
+  public Object asSerializableValue() {
+    return values.stream().map(CbasValue::asSerializableValue).toList();
+  }
+
+  @Override
+  public long countFiles() {
+    return values.stream().map(CbasValue::countFiles).reduce(0L, Long::sum);
```

Comment:
Java's stream APIs have support for numeric types that provides unboxed access. It's more efficient and natively supports things like `sum`.
```suggestion
    return values.stream().mapToLong(CbasValue::countFiles).sum();
```

Reply:
Very cool!